### ISSUE TITLE
Requesting clarification on whether implicit-sgd can operate on one data at a time. 

### ISSUE THREAD

**Issue by craigphicks** [ptoulis/implicit-sgd#4]
The SVG algorithm can operate one data at a time, and although mini-batch processing may offer a constant speed-up by enabling vector processing, it is not required by the algorithm.  
I would like to clarify whether or not implicit-SVG can also operate on one data at a time, and if the answer is affirmative, if it requires extra memory, and if so the size of that extra memory with respect to number of weights.  

I am looking at your paper "Implicit stochastic gradient descent", Algorithm 4.1, so I should be able to answer my own question, but unfortunately due to my limited layman's grasp, I cannot.

For reference, on the general topic of line optimization, I have found [this](https://stats.stackexchange.com/a/321649) general qualitative comment on the subject.  To quote:

> Vanilla gradient descent can be made more reliable using line searches; I've written algorithms that do this and it makes for a very stable algorithm (although not necessarily fast).

> However, it makes almost no sense to do a line search for stochastic gradient methods. The reason I say this is that if we do a line search based on minimizing the full loss function, we've immediately lost one of the main motivations for doing stochastic methods; we now need to compute the full loss function for each update, which typically has computational cost comparable to computing the full first derivative. Given that we wanted to avoid computing the full gradient because of computational costs, it seems very unlikely that we want be okay with computing the full loss function.

> Alternatively, you might think of doing something like a line search based on your randomly sampled data point. However, this isn't a good idea either; this will tell you nothing about whether you have stepped too far (which is the main benefit of line searches). For example, suppose you are performing logistic regression. Then each outcome is simply a 0 or 1, and for any single sample, we trivially get perfect separation so the optimal solution for our regression parameters based on the sample of 1 is trivially −∞
or ∞ by the Hauck Donner effect. That's not good.

> EDIT  @DeltaIV points out that this also applies to mini-batch, not just individual samples.
 
Does the above comment intersect with *implcit-svg*, and if so, how to resolve it?
